### PR TITLE
add ut for util/metrics

### PR DESCRIPTION
**Pull request by my-git9**
**What type of PR is this?**
add ut for util/metrics

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/util/metrics/metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		start time.Time
+	}{
+		{
+			name:  "testMicroseconds",
+			start: time.Date(2022, 10, 1, 7, 8, 9, 0, time.Local),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := DurationInMicroseconds(tt.start)
```
Comment (Poor12) [inline pkg/util/metrics/metrics_test.go:48]: I'm not sure that for these functions about time, such a test method is sufficient.

File: pkg/util/metrics/metrics_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+)
+
+func TestGetResultByError(t *testing.T) {
+	tests := []struct {
+		name  string
+		error error
+		want  string
+	}{
+		{
+			name:  "testnoerr",
```
Comment (Poor12) [inline pkg/util/metrics/metrics_test.go:17]: Test case names can be more specific.

File: pkg/util/metrics/metrics_test.go (hunk 1 above)
Comment (my-git9) [reply pkg/util/metrics/metrics_test.go:48]: I have the same thought, but this function can't get a definite value for comparison. Can you give me some advice? Thanks!

**Comment by XiShanYongYe-Chang** [comment]
ping @Poor12

File: pkg/util/metrics/metrics_test.go (hunk 1 above)
Comment (Poor12) [reply pkg/util/metrics/metrics_test.go:48]: /cc @XiShanYongYe-Chang for opinion.

Comment (XiShanYongYe-Chang) [reply pkg/util/metrics/metrics_test.go:48]: Personally, I think it makes no sense to test this function. I suggest removing unused functions.
Hi @my-git9 How do you think?

Comment (my-git9) [reply pkg/util/metrics/metrics_test.go:48]: Do you mean we want to delete unused functions from "pkg/util/metrics/metrics.go"? It seems that none of the functions inside were used

Comment (XiShanYongYe-Chang) [reply pkg/util/metrics/metrics_test.go:48]: > Do you mean we want to delete unused functions from "pkg/util/metrics/metrics.go"? 

Yes.

Those two methods seem not used:

https://github.com/karmada-io/karmada/blob/594ad9f44b67d199354ce375c9df43adcb6187e9/pkg/util/metrics/metrics.go#L22-L30

Comment (my-git9) [reply pkg/util/metrics/metrics_test.go]: Has beed updated, please review again, thx

Comment (XiShanYongYe-Chang) [reply pkg/util/metrics/metrics_test.go]: Thanks~ There is a lint error.

File: pkg/util/metrics/metrics_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			start: time.Date(2022, 10, 1, 7, 8, 9, 0, time.Local),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := DurationInSeconds(tt.start)
+			fmt.Printf("DurationInSeconds() = %v\n", got)
+		})
+	}
+}
```
Comment (XiShanYongYe-Chang) [inline pkg/util/metrics/metrics_test.go:52]: Hi @my-git9, This test does not judge the test result as expected and can be removed.

Comment (my-git9) [reply pkg/util/metrics/metrics_test.go]: Updated

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve
Thanks.

**Pull request merged**